packages/client/src/runtime/core/init/resolveDatasourceUrl.ts

```diff
@@ -29,6 +29,16 @@ export function resolveDatasourceUrl({
 
   // env var is set for use but url is undefined
   if (datasourceUrl?.fromEnvVar !== undefined && resolvedUrl === undefined) {
+    if (TARGET_ENGINE_TYPE === 'edge') {
+      throw new PrismaClientInitializationError(
+        `error: Environment variable not found: ${datasourceUrl.fromEnvVar}.
+Only \`process.env\` and \`globalThis\` can be read, but not \`.env\`.
+
+To solve this, provide it directly: https://pris.ly/d/datasource-url`,
```

Comment:
```suggestion
      throw new PrismaClientInitializationError(
        `error: Environment variable not found: ${datasourceUrl.fromEnvVar}.
When using Prisma edge client, environment variables are read only from \`process.env\` or \`globalThis\` globals, \`.env\` file is not accessible.

To solve this, provide it directly: https://pris.ly/d/datasource-url`,
```

Just a suggestion, feel free to ignore.